packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/LogObfuscator.kt

```diff
@@ -0,0 +1,248 @@
+            input.contains("access_token") &&
+            input.contains("refresh_token")
+        ) {
+            val beforeAccessToken = input.substringBefore(""""access_token"""")
+            val afterRefreshToken = input.substringAfter(""""user_id":""")
+            """$beforeAccessToken"access_token":"***","refresh_token":"***","user_id":$afterRefreshToken"""
+        } else {
+            var obfuscatedString = input
+            val entries: Set<Map.Entry<Regex, String>> = patternReplacementMap.entries
+            for (entry in entries) {
+                val pattern = entry.key
+                obfuscatedString = pattern.replace(obfuscatedString, entry.value)
+            }
+            obfuscatedString
+        }
+    }
+}
+
+// Replacer for custom function arguments. It combs the request and hides both the parameters that
+// are send and any possible received result
+internal object CustomFunctionPatternReplacer : LogReplacer {
+    override fun findAndReplace(input: String): String {
+        val (pattern, replacement) = when {
+            input.contains("REQUEST: ") ->
+                """("arguments"):\[.*]""".toRegex() to """"arguments":[***]"""
+            input.contains("RESPONSE: 200 OK") ->
+                """BODY START\n.*\nBODY END""".toRegex() to "BODY START\n***\nBODY END"
+            else -> return input
+        }
+        return pattern.replace(input, replacement)
+    }
+}
+
+internal object LogObfuscatorImpl : HttpLogObfuscator {
+
+    private val urlRegex =
+        Regex("https?:\\/\\/(www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()!@:%_\\+.~#?&\\/\\/=]*)")
+
+    private val features: Collection<ObfuscatorFeature> = setOf(
+        ObfuscatorFeature.FEATURE_EMAIL_REGISTER,
+        ObfuscatorFeature.FEATURE_EMAIL_LOGIN,
+        ObfuscatorFeature.FEATURE_API_KEY_CREATE,
+        ObfuscatorFeature.FEATURE_API_KEY_LOGIN,
+        ObfuscatorFeature.FEATURE_APPLE_LOGIN,
+        ObfuscatorFeature.FEATURE_FACEBOOK_LOGIN,
+        ObfuscatorFeature.FEATURE_GOOGLE_LOGIN,
+        ObfuscatorFeature.FEATURE_JWT_LOGIN,
+        ObfuscatorFeature.FEATURE_CUSTOM_FUNCTION_REQUEST
+    )
+
+    private val regexReplacerMap: Map<String, LogReplacer> = defaultFeatureToReplacerMap
+
+    override fun obfuscate(input: String): String {
+        features.forEach { feature ->
+            urlRegex.find(input)?.let { matchResult ->
+                val url = matchResult.value
+                if (url.contains(feature.urlPath)) {
+                    val patternReplacer = regexReplacerMap[feature.urlPath]
+                    return patternReplacer?.findAndReplace(input) ?: input
+                }
+            }
+        }
+        return input
+    }
+}
+
+// Features that will be obfuscated by default. It uses the `urlPath` to tell the replacer when to
+// find the patterns in the logger
+internal enum class ObfuscatorFeature(internal val urlPath: String) {
```

Comment:
Again seems redundant compared to just use the `urlPath` of these as keys in `defaultFeatureToReplacerMap`